### ISSUE TITLE
2022.11 regression on kindle PW3: swipe to unlock!?

### ISSUE THREAD

**Issue by SnarkBoojum** [koreader/koreader#9883]
* KOReader version: 2022.11
* Device: Kindle PW3

With the previous version of koreader, when I opened my reader (magnetic cover), it directly woke up to the current page and I could start reading.

With the current version, the screen is refreshed from the sleep screen to the sleep+"Swipe to unlock" message screen, but in fact, if I touch the screen, it flips the page: it really was already unlocked, but just didn't display the current page.

I checked the settings for refresh, but it only covers refreshing every x page or each chapter start, nothing about doing it when going out of sleep.

**Comment by NiLuJe** [comment]
That's a special offers device, right?

**Comment by SnarkBoojum** [comment]
If you mean a device where advertisement is displayed on the sleep screen, yes.

But it's an always-offline device where the previous version of koreader displayed the page to read on cover opening and where the new version changes from the sleep screen to sleep screen+"swipe to unlock" message, so I really think there's a small bug to fix. A simple refresh might be enough...

**Comment by NiLuJe** [comment]
Which version were you running previously?

**Comment by SnarkBoojum** [comment]
Version 2022.10.

**Comment by NiLuJe** [comment]
Huh. I'll have to double-check on my old SO PW1.

**Comment by NiLuJe** [comment]
There was one, but it was in an earlier release, hence my question ;).

The one that landed in 2022.11 is... sneakier to interpret, hence the need to see for myself ;).

**Comment by SnarkBoojum** [comment]
I kicked 2022.11 and re-installed 2022.10: I can confirm once again that 2022.10 didn't have the problem.

**Comment by SnarkBoojum** [comment]
Excellent ; thanks!

I'm looking forward to 2022.12!

**Comment by SnarkBoojum** [comment]
Ah, I re-installed 2022.11 and applied your patch by hand: the longer timeout is too long, I'll revert that part.

**Comment by NiLuJe** [comment]
That was tested in the worst conditions possible for an eInk screen:

* The slowest device I own (a PW1)
* It's winter over here, so, it's cold
* At night, so it was even colder
* After an actual suspend, which makes resumes a tiny bit slower

TL;DR: Yes, it's *very* conservative. But there's no silver-bullet, it depends on too many factors, the only constant is that *some* delay is necessary, because SO does... stuff on resume.

So it was kind of a lesser evil compared to making the delay too short, and getting an extra refresh that would... just refresh the SO ad ;o).

**Comment by NiLuJe** [comment]
(That's a slight hint that'd you'd have a *much* better experience shelling out twenty bucks to your evil overlords to get rid of SO ;o).)
